src/compiler/program.rs

```diff
@@ -101,27 +185,31 @@ pub fn create_program(root_names_or_options: CreateProgramOptions) -> impl Progr
             host: &host,
             current_directory: &current_directory,
         };
-        for_each(root_names, &mut |name, _index| {
+        for_each(root_names, |name, _index| {
             process_root_file(&mut helper_context, name);
             Option::<()>::None
         });
 
         files = processing_other_files_present;
-        println!("{:?}", files);
+        println!("files: {:?}", files);
         processing_other_files = None;
     }
 
     ProgramConcrete::new(files)
 }
 
+fn filter_semantic_diagnostics(diagnostic: Vec<Rc<Diagnostic>>) -> Vec<Rc<Diagnostic>> {
+    diagnostic
+}
+
 fn process_root_file(helper_context: &mut CreateProgramHelperContext, file_name: &str) {
     process_source_file(helper_context, &normalize_path(file_name));
 }
 
 fn get_source_file_from_reference_worker(
     file_name: &str,
-    get_source_file: &mut dyn FnMut(&str) -> Option<Rc<SourceFile>>,
-) -> Option<Rc<SourceFile>> {
+    get_source_file: &mut dyn FnMut(&str) -> Option<Rc<Node>>,
```

Comment:
Again here maybe this should be generic over the closure type rather than taking a reference to a `dyn` closure type